app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -28,6 +28,24 @@ def parent_cluster
   alias_method :owning_cluster, :parent_cluster
   alias_method :ems_cluster, :parent_cluster
 
+  def disconnect_storage(s = nil)
+    stores = s.nil? ? ([storage] + storages).compact.uniq : [s]
+    log_text = stores.collect { |x| "#{ui_lookup(:table => "storages")} [#{x.name}] id [#{x.id}]" }.join(", ")
+
+    disconnect = true
+    for storage in stores
+      if storage.disks.any? { |disk| self.disks.include?(disk) }
+        disconnect = false
+      end
+    end
+
+    if disconnect
+      self.storage = nil
+      self.storages = []
+      save
+    end
```

Comment:
Since a lot of this is duplicated by `VmOrTemplate#disconnect_storage` could you just check for disks in this method and call `super` if the disks are gone?